sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/NetworkStateHelper.java

```diff
@@ -149,81 +144,54 @@ private IntentFilter getOldIntentFilter() {
      * @return true for connected, false for disconnected.
      */
     public synchronized boolean isNetworkConnected() {
-        return mNetworkType != null || !mAvailableNetworks.isEmpty();
-    }
 
-    /**
-     * Handle network available update on API level >= 21.
-     */
-    private synchronized void onNetworkAvailable(Network network) {
-        AppCenterLog.debug(LOG_TAG, "Network available netId: " + network);
-        mAvailableNetworks.add(network);
-        AppCenterLog.debug(LOG_TAG, "Available networks netIds: " + mAvailableNetworks);
+        /* The actual current state should be checked here instead of using a value based on connection change events. */
+        try {
+            NetworkInfo info = mConnectivityManager.getActiveNetworkInfo();
```

Comment:
No this is inaccurate and caused issues in the past, we had to move away from this call. Please revert this change.

Reply:
Maybe keep it as a fallback? Because rely just on connection change events is inaccurate too

Reply:
This API has already been proven to not work correctly like returning "blocked" state while the network was up. It is also deprecated. It also does not work correctly when there is more than 1 network as mentioned in the past by a customer who worked on a network library.

Reply:
it's not deprecated, but OK, I got it. Thanks for the info

Reply:
Ah yeah its `getAllNetworkInfo()` that is deprecated indeed.